core/src/main/scala/sttp/tapir/monad/MonadError.scala

```diff
@@ -11,31 +11,33 @@ trait MonadError[F[_]] {
   def flatMap[T, T2](fa: F[T])(f: T => F[T2]): F[T2]
 
   def error[T](t: Throwable): F[T]
-  protected def handleWrappedError[T](rt: F[T])(h: PartialFunction[Throwable, F[T]]): F[T]
-  def handleError[T](rt: => F[T])(h: PartialFunction[Throwable, F[T]]): F[T] = {
-    Try(rt) match {
-      case Success(v)                     => handleWrappedError(v)(h)
-      case Failure(e) if h.isDefinedAt(e) => h(e)
-      case Failure(e)                     => error(e)
-    }
-  }
+  def handleError[T](rt: => F[T])(h: PartialFunction[Throwable, F[T]]): F[T]
```

Comment:
Shouldn't the parameter now become strict? I don't think we use anywhere the fact that it's by-name?

Reply:
Actually it's what has been suggested here https://github.com/softwaremill/tapir/issues/658#issuecomment-658697681 

I thought it gives more control in case some instance wanted to deal with catching exception from evaluation on their own.

However both works for me, so just say your preference and I'll change it ;)

Reply:
We're already dealing with exceptions during effect construction elsewhere, so I suppose this doesn't have any utility (as demonstrated by the current code ;)). So I'd remove it.